Added note on renovate vs dependabot (#7378)
jenkins-infra/jenkins.io: content/doc/developer/publishing/releasing-cd.adoc

=== Configure Dependabot

> NOTE: If you have a `.github/renovate.json`, do not configure Dependabot. link:/blog/2023/09/20/renovate-bot-probe-blog[For more details]

Reviewer: ```suggestion
NOTE: If you have a `.github/renovate.json`, do not configure Dependabot. Refer to the following blog post link:/blog/2023/09/20/renovate-bot-probe-blog[for more details].
```
Adjusting the text so that it is a full sentence with a full stop.

Reviewer 2: I can't see how this blog post is related it should not be linked here.

Something like:
```suggestion
NOTE: If you have link:https://docs.renovatebot.com/[Renovate] configured (normally in `renovate.json` or `.github/renovate.json`) do not configure Dependabot, as long as your renovate configuration isn't explicitly enabling a single manager renovate will automatically start updating your new dependencies that are being added here.
```

or a shorter one:

```suggestion
NOTE: If you have link:https://docs.renovatebot.com/[Renovate] configured (normally in `renovate.json` or `.github/renovate.json`) do not configure Dependabot.
```

> NOTE: If you have a `.github/renovate.json`, do not configure Dependabot. link:https://www.jenkins.io/blog/2023/09/20/renovate-bot-probe-blog[For more details]

Reviewer 3: We avoid placing the www.jenkins.io part into the URL inside the site because it makes it more difficult to test the site from a local build.
```suggestion
NOTE: If you have a `.github/renovate.json`, do not configure Dependabot. link:/blog/2023/09/20/renovate-bot-probe-blog[For more details]
```

> Note:: If you have a `.github/renovate.json`, do not configure Dependabot. link:https://www.jenkins.io/blog/2023/09/20/renovate-bot-probe-blog[For more details]

Reviewer 4: We shouldn't be too categorical; perhaps we could propose a discussion with other maintainers to choose the right tool for handling dependency updates.

The long-term goal is to ensure that maintainers are comfortable with the tools they use.

If a maintainer is working alone on a project and prefers Dependabot to Renovate, we should allow them to choose Dependabot and discontinue the use of Renovate.

Reviewer 2: I think this should section should really be configure a dependency update tool either renovate or dependabot. 

When this was written dependabot was the only one in use. Since then renovate has been used a lot more as it is way more powerful

Reviewer 4: I think we all have biases when it comes to choosing our dependency update tool.

Do we really need a "more powerful" tool when a simple one does the trick?

I believe it depends on several factors:
1. The complexity of updating certain dependencies
2. The maintainers' skills
3. The willingness of maintainers to learn a new tool

Reviewer 2: > Do we really need a "more powerful" tool when a simple one does the trick?

Yes, dependabot doesn't scale. Its fine for simple cases.

I'm not saying prefer renovate here just to give the option. 

In terms of this pull request I think removing the blog link and rewording to:

> If you have [renovate](https://docs.renovatebot.com/) configured there is no need to configure dependabot

Would be good to work `renovate.json` in, keeping in mind it can be in a number of places and people will put it in different ones

> Note:: If you have a `.github/renovate.json`, do not configure Dependabot. link:https://www.jenkins.io/blog/2023/09/20/renovate-bot-probe-blog[For more details]

Reviewer 5: use an admonition?

```suggestion
NOTE: If you have a `.github/renovate.json`, do not configure Dependabot. link:https://www.jenkins.io/blog/2023/09/20/renovate-bot-probe-blog[For more details]
```

> NOTE: If you have a `.github/renovate.json`, do not configure Dependabot. Refer to the following blog post link:/blog/2023/09/20/renovate-bot-probe-blog[for more details].

Reviewer 2: I can't see how this blog post is related it should not be linked here.

Something like:
```suggestion
NOTE: If you have link:https://docs.renovatebot.com/[Renovate] configured (normally in `renovate.json` or `.github/renovate.json`) do not configure Dependabot, as long as your renovate configuration isn't explicitly enabling a single manager Renovate will automatically start updating your new dependencies that are being added here.
```

or a shorter one:

```suggestion
NOTE: If you have link:https://docs.renovatebot.com/[Renovate] configured (normally in `renovate.json` or `.github/renovate.json`) do not configure Dependabot.
```
